xmppserver/src/main/java/org/jivesoftware/util/ListPager.java

```diff
@@ -83,6 +96,21 @@ public ListPager(final HttpServletRequest request, final HttpServletResponse res
      * @param additionalFormFields 0 or more input field identifiers (<strong>NOT form field names</strong>) to include in requests for other pages
      */
     public ListPager(final HttpServletRequest request, final HttpServletResponse response, final List<T> items, final Predicate<T> filter, final String... additionalFormFields) {
+        this(request, response, items, filter, 0, false, additionalFormFields);
+    }
+
+    /**
+     * Creates a unfiltered list pager.
+     *
+     * @param request              the request object for the page in question
+     * @param response             the response object for the page in question
+     * @param items                the complete list of items to display on the page
+     * @param sortColumnNumber     the index of the column (0-based) of which values are to be ordered
+     * @param sortDescending       true if the column is to be in descending order, false for ascending order.
+     * @param filter               the filter to apply to the complete list
+     * @param additionalFormFields 0 or more input field identifiers (<strong>NOT form field names</strong>) to include in requests for other pages
+     */
+    public ListPager(final HttpServletRequest request, final HttpServletResponse response, final List<T> items, final Predicate<T> filter, final int sortColumnNumber, final boolean sortDescending, final String... additionalFormFields) {
```

Comment:
I wonder if `sortDescending `should be an `Boolean `rather than a `boolean`, to allow some pages that support sorting to also specify "I don't care" in the same constructor?

Reply:
(I can see this being useful on, e.g., the page that shows users, but for servers backed by large corporate LDAP servers they may not want to sort the whole lot unless they have to)

Reply:
For those, the overloaded constructors (the ones that does not have the `final int sortColumnNumber` and `sortDescending` parameters) are available. The implementation currently does not need a null-value, as 'I do not want ordering' is expressed by using an invalid column number value. That one might instead be a `null`, perhaps. However, as users of that particular overloaded constructors _want_ ordering of a column, I don't think it matters to much.

Reply:
Expanding on my thinking;

The first time you hit the users page, you wouldn't want sorting - to avoid large numbers of users throwing things off.

However, installations with small numbers of users may which to sort by name, which they could do by clicking on a column.

Rather than the code backing this relying on calling one of two constructors depending on if the sorting was selected or not, a single constructor with a nullable param might help. But perhaps YAGNI :)

Reply:
The actual sorting is not done in ListPager, it's done by the caller. Currently, that defaults to a non-null value, but I assume that the vast majority of people prefer to have some kind of order.

We can optimize this later, but we're not introducing a new overhead here: at least the pubsub node stuff already had sorting applied. We are now only expanding the possible columns to sort on.